From c7552038b51ceddb24459bf0bad2d40f16ab34b9 Mon Sep 17 00:00:00 2001
From: Kyle Brenneman <kbrenneman@nvidia.com>
Date: Wed, 19 Sep 2018 15:23:49 -0600
Subject: [PATCH 3/4] GLdispatch: Keep the mutex when calling the vendor's
 threadAttach callback

In __glDispatchCheckMultithreaded, don't unlock the mutex until after calling
stubCurrentPatchCb->threadAttach.

Without that lock, another thread could come along and change
stubCurrentPatchCb out from under it.
---
 src/GLdispatch/GLdispatch.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/GLdispatch/GLdispatch.c b/src/GLdispatch/GLdispatch.c
index 09fc75d..386b543 100644
--- a/src/GLdispatch/GLdispatch.c
+++ b/src/GLdispatch/GLdispatch.c
@@ -764,11 +764,11 @@ void __glDispatchCheckMultithreaded(void)
                 _glapi_set_multithread();
             }
         }
-        UnlockDispatch();
 
         if (stubCurrentPatchCb != NULL && stubCurrentPatchCb->threadAttach != NULL) {
             stubCurrentPatchCb->threadAttach();
         }
+        UnlockDispatch();
     }
 }
 
-- 
2.19.1

